kyo-data/shared/src/main/scala/kyo/Record.scala

```diff
@@ -125,6 +126,23 @@ object Record:
       */
     val empty: Record[Any] = Record[Any](Map())
 
+    private def unsafeFrom[Fields: AsFields](map: Map[Field[?, ?], Any]): Record[Fields] = Record(map)
```

Comment:
is this necessary to make type inference work?

Reply:
This is because `stage` is `inline` method. When I try to call private constructor of `Record` inside it, I get the error: 
`Implementation restriction: cannot use private constructors in inline methods`